src/Connection.php

```diff
@@ -75,7 +75,7 @@ public function getMailbox($name)
             throw new MailboxDoesNotExistException($name);
         }
 
-        return new Mailbox($this->server . imap_utf7_encode($name), $this);
+        return new Mailbox($this->server . mb_convert_encoding($name, "UTF7-IMAP", "UTF-8"), $this);
```

Comment:
Implemented in https://github.com/ddeboer/imap/pull/193 due to the heavy refactored code.
Thank you for your help